python/pyspark/sql/dataframe.py

```diff
@@ -553,10 +594,10 @@ def repartition(self, numPartitions, *cols):
         +---+-----+
         |age| name|
         +---+-----+
-        |  2|Alice|
         |  5|  Bob|
-        |  2|Alice|
         |  5|  Bob|
+        |  2|Alice|
+        |  2|Alice|
```

Comment:
It seems the order is varied dependent on how we create the dataframe (as I saw the related JIRA I can point out if anyone wants). I haven't looked into this deeper as this variant seems not related to the tests in `repartition` API.